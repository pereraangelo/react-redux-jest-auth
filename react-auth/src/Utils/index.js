import moment from "moment";

export const getGreetingTime = () => {
  let currentTime = new Date();

  const splitAfternoon = 12; // 24hr time to split the afternoon
  const splitEvening = 15; // 24hr time to split the evening
  const currentHour = moment(currentTime).format("HH");

  if (currentHour >= splitAfternoon && currentHour <= splitEvening) {
    // Between 12 PM and 5PM
    return "Good Afternoon";
  } else if (currentHour >= splitEvening) {
    // Between 5PM and Midnight
    return "Good Evening";
  }
  // Between Dawn and noon
  return "Good Morning";
};



export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const simulateChangeOnInput = (wrapper, inputSelector, newValue) => {
  let input = findByTestAtrr(wrapper, inputSelector);
  input.simulate("change", {
    target: { name: inputSelector, value: newValue },
  });

  return findByTestAtrr(wrapper, inputSelector);
};


export const simulateCheckBox = (wrapper, inputSelector, newValue) => {
    let input = findByTestAtrr(wrapper, inputSelector);
    input.simulate('change', {target: {name:inputSelector,checked: newValue}});
  
    return findByTestAtrr(wrapper, inputSelector);
  };

  