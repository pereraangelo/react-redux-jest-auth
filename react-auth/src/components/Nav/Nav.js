import React from "react";
import { connect } from "react-redux";
import * as actions from "../../actions/authActions";
import { getGreetingTime } from "../../Utils";



export function Nav(props) {
  const { logout, isAuthenticated, userName } = props;
  return (
    <div className="header">
      <div className="logout">
        {isAuthenticated && (
          <>
            <div className="mx-3"  data-test="greeting">{getGreetingTime() + " " + userName}</div>
            <div className="logoutBtn"  data-test="logout" onClick={logout}>
              LogOut
            </div>
          </>
        )}
      </div>
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    userName: state.auth.user.userName,
    isAuthenticated: state.auth.isAuthenticated,
   
  };
};

export default connect(mapStateToProps, actions)(Nav);
