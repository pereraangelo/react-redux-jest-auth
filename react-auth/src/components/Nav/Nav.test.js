
import React from "react";
import ReactDOM from 'react-dom'
import {Nav} from "./Nav";
import {MemoryRouter} from "react-router";
import renderer from 'react-test-renderer'
import { shallow } from "enzyme";
import { findByTestAtrr } from "../../Utils";

const defaultProps={
    logout: jest.fn(),
    isAuthenticated:false,
    userName:""
  }
  
  
  
 
describe('NavBar component',()=>{
    const wrapper = shallow(<Nav {...defaultProps} />);
    it('Nav should load without crahing',()=>{
        const div = document.createElement('div');
        ReactDOM.render(<Nav/>, div);
    });
    
    
    it('login component matches the snapshot',()=>{
        const tree=renderer.create(<MemoryRouter ><Nav/></MemoryRouter>).toJSON()
        expect(tree).toMatchSnapshot()
    });
      
      
    it('shows logout button when user logged in',()=>{

        wrapper.setProps({ isAuthenticated: true });
        const logoutBtn=findByTestAtrr(wrapper, 'logout')
        expect(logoutBtn.length).toEqual(1)
 
    })

      
    it('greets user when user logged in',()=>{
        
        wrapper.setProps({ isAuthenticated: true });
        const logoutBtn=findByTestAtrr(wrapper, 'greeting')
        expect(logoutBtn.length).toEqual(1)
 
    })

})

