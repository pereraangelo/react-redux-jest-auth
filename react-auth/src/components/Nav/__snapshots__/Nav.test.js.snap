// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component login component matches the snapshot 1`] = `
<div
  className="header"
>
  <div
    className="logout"
  />
</div>
`;
